fink_filters/filter_anomaly_notification/filter.py

```diff
@@ -0,0 +1,72 @@
+from pyspark.sql.functions import when, lit
+import numpy as np
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+import pandas as pd
+import filter_utils
+
+
+
+def anomaly_notification_(df, threshold=10) -> pd.Series:
+    """
+    Create event notifications with a high anomaly_score value
+    Parameters
+    ----------
+    df : Spark DataFrame with column :
+        objectId : unique identifier for this object
+        lc_features: Dict of dicts of floats.
+            Keys of first dict - filters (fid),
+            keys of inner dicts - names of features
+        rb: RealBogus quality score
+        anomaly_score: Anomaly score
+        timestamp : UTC time
+    threshold : Number of notifications (10 by default)
+
+    Returns
+    ----------
+    out: pandas.Series of bool
+    Return a Pandas DataFrame with the appropriate flag:
+    false for bad alert, and true for good alert.
+    Examples
+    ----------
+    >>> from fink_utils.spark.utils import concat_col
+    >>> from fink_science.ad_features.processor import extract_features_ad
+    >>> from fink_science.anomaly_detection.processor import anomalys_score
+    >>> df = spark.read.format('parquet').load('datatest')
+    >>> what = ['magpsf', 'jd', 'sigmapsf', 'fid']
+    >>> prefix = 'c'
+    >>> what_prefix = [prefix + i for i in what]
+    >>> for colname in what:
+    ...    df = concat_col(df, colname, prefix=prefix)
+    >>> df = df.withColumn('lc_features', extract_features_ad(*what_prefix, 'objectId'))
+    >>> df = df.withColumn("anomaly_score", anomaly_score("lc_features"))
+    >>> df_proc = df.select('objectId', 'candidate.ra',
+    >>>                     'candidate.dec', 'candidate.rb',
+    >>>                     'anomaly_score', 'timestamp')
+    >>> mask = anomaly_notification_(df_proc)
+    >>> print(sum(mask))
+    """
+    med = df.select('anomaly_score').approxQuantile('anomaly_score', [0.5], 0.25)
+    df_filtred = df.sort(['anomaly_score'], ascending=True).limit(min(df.count(), threshold))
```

Comment:
No need for `min` -- by default Spark will look for `threshold`, otherwise it will stop before. Moreover this `min` forced you to compute the `df.count()` which was an extra overhead.

The proposed skeleton simplifies it.

Reply:
Understood, thank you!